primitives/core/src/traits.rs

```diff
@@ -187,20 +192,143 @@ pub trait BareCryptoStore: Send + Sync {
+	}
+
+	pub fn sr25519_generate_new(
+		&self,
+		id: KeyTypeId,
+		seed: Option<&str>,
+	) -> Result<sr25519::Public, Error> {
+		block_on(self.0.sr25519_generate_new(id, seed))
+	}
+
+	pub fn ed25519_public_keys(&self, id: KeyTypeId) -> Vec<ed25519::Public> {
+		block_on(self.0.ed25519_public_keys(id))
+	}
+
+	pub fn ed25519_generate_new(
+		&self,
+		id: KeyTypeId,
+		seed: Option<&str>,
+	) -> Result<ed25519::Public, Error> {
+		block_on(self.0.ed25519_generate_new(id, seed))
+	}
+
+	pub fn ecdsa_public_keys(&self, id: KeyTypeId) -> Vec<ecdsa::Public> {
+		block_on(self.0.ecdsa_public_keys(id))
+	}
+
+	pub fn ecdsa_generate_new(
+		&self,
+		id: KeyTypeId,
+		seed: Option<&str>,
+	) -> Result<ecdsa::Public, Error> {
+		block_on(self.0.ecdsa_generate_new(id, seed))
+	}
+
+	pub fn insert_unknown(&self, key_type: KeyTypeId, suri: &str, public: &[u8]) -> Result<(), ()> {
+		block_on(self.0.insert_unknown(key_type, suri, public))
+	}
+
+	pub fn supported_keys(
+		&self,
+		id: KeyTypeId,
+		keys: Vec<CryptoTypePublicPair>
+	) -> Result<Vec<CryptoTypePublicPair>, Error> {
+		block_on(self.0.supported_keys(id, keys))
+	}
+
+	pub fn keys(&self, id: KeyTypeId) -> Result<Vec<CryptoTypePublicPair>, Error> {
+		block_on(self.0.keys(id))
+	}
+
+	pub fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool {
+		block_on(self.0.has_keys(public_keys))
+	}
+
+	pub fn sign_with(
+		&self,
+		id: KeyTypeId,
+		key: &CryptoTypePublicPair,
+		msg: &[u8],
+	) -> Result<Vec<u8>, Error> {
+		block_on(self.0.sign_with(id, key, msg))
+	}
+
+	pub fn sign_with_any(
+		&self,
+		id: KeyTypeId,
+		keys: Vec<CryptoTypePublicPair>,
+		msg: &[u8]
+	) -> Result<(CryptoTypePublicPair, Vec<u8>), Error> {
+		block_on(self.0.sign_with_any(id, keys, msg))
+	}
+
+	pub fn sign_with_all(
+		&self,
+		id: KeyTypeId,
+		keys: Vec<CryptoTypePublicPair>,
+		msg: &[u8],
+	) -> Result<Vec<Result<Vec<u8>, Error>>, ()> {
+		block_on(self.0.sign_with_all(id, keys, msg))
+	}
+
+	/// Generate VRF signature for given transcript data.
+	///
+	/// Receives KeyTypeId and Public key to be able to map
+	/// them to a private key that exists in the keystore which
+	/// is, in turn, used for signing the provided transcript.
+	///
+	/// Returns a result containing the signature data.
+	/// Namely, VRFOutput and VRFProof which are returned
+	/// inside the `VRFSignature` container struct.
+	///
+	/// This function will return an error in the cases where
+	/// the public key and key type provided do not match a private
+	/// key in the keystore. Or, in the context of remote signing
+	/// an error could be a network one.
+	pub fn sr25519_vrf_sign(
+		&self,
+		key_type: KeyTypeId,
+		public: &sr25519::Public,
+		transcript_data: VRFTranscriptData,
+	) -> Result<VRFSignature, Error> {
+		block_on(self.0.sr25519_vrf_sign(key_type, public, transcript_data))
+	}
+}
+
+impl From<Arc<dyn CryptoStore>> for SyncCryptoStore {
+    fn from(store: Arc<dyn CryptoStore>) -> Self {
```

Comment:
Spaces.

Reply:
I am curious, how you see this in the github diff?

Reply:
FIrst you see that the line below is intended differently and than I also have a plugin for this.